### PR TITLE
#TP-26 Creating redux setup

### PR DESCRIPTION
**Pull request by sezeranoJchrisostome**
### What does this PR do?

> This pr is for the initial redux setup

### Description 

> I made a slice that is counter with the count state. It is either. incremented, decremented, or incremented by any value in the payload 

### How you can manually test it.

> clone this rep if you have not already `git clone https://github.com/atlp-rwanda/phantom-fe-techode.git`

1. navigate to phantom-fe-techode
2. `git fetch`
3. `git checkout ch-setup-redux-TP-26`
4. `npm install`
5. `npm start`

Try to increment by using the increment button and decrement as well,  you can also increment it by value which is currently set to 3

### Related trello board

> [#TP-26](https://trello.com/c/WZacMsLQ/26-setup-redux)

### Screenshots
 
Screenshot (31)

Screenshot (32)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,13 +19,16 @@
   },
   "homepage": "https://github.com/atlp-rwanda/phantom-fe-techode#readme",
   "dependencies": {
+    "@reduxjs/toolkit": "^1.7.2",
```
Comment (renepromesse) [inline package.json:22]: remove these unused packages

File: src/redux/reducers/countReducer.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { ActionTypes } from "../constants/actionType"
+
+const initialState = {
+    count: 0
+}
+
+export const counterReducer = (state = initialState , { type , payload}) =>{
```
Comment (renepromesse) [inline src/redux/reducers/countReducer.js:7]: try to implement the DRY features

File: src/redux/actions/counterActions.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { ActionTypes } from "../constants/actionType"
```
Comment (renepromesse) [inline src/redux/actions/counterActions.js:1]: destruct the ActionTypes object

**Review by renepromesse — CHANGES_REQUESTED** [review]
finish up on this mentioned feedback and push again

**Review by chanceDira — CHANGES_REQUESTED** [review]
Change mode

**Review by chanceDira — APPROVED** [review]

**Review by chanceDira — APPROVED** [review]
Great work @sezeranoJchrisostome

**Review by renepromesse — APPROVED** [review]

**Pull request merged**